src/Microsoft.Cci.Extensions/Writers/CSharp/CSDeclarationWriter.Methods.cs

```diff
@@ -206,7 +206,10 @@ private void WriteMethodBody(IMethodDefinition method)
                 Write("throw new ");
                 if (_forCompilationIncludeGlobalprefix)
                     Write("global::");
-                Write("System.PlatformNotSupportedException(); ");
+                if(_platformNotSupportedExceptionMessage == null)
+                    Write("System.PlatformNotSupportedException();");
+                else
+                    Write($"System.PlatformNotSupportedException(\"{_platformNotSupportedExceptionMessage}\"); ");
```

Comment:
> Wouldn't we also need to build resource generation into this task? I assume it currently doesn't bother including SR at all (but I could easily be wrong)

This already works today.  The PNSE infra just generates source that gets passed to compile.  SR generation works the same way.

@danmosemsft it's pretty trivial to implement support for SR now.  Just choose a pattern and use that.  Seems like it'd be better to keep that 5% low rather than add to it, right?